readme.md

```diff
@@ -16,20 +16,51 @@ A manual is available [online](https://cppcheck.sourceforge.io/manual.pdf).
 
 ## Donate CPU
 
-Cppcheck is a hobby project with limited resources. You can help us by donating CPU (1 core or as many as you like). It is simple:
+Cppcheck is a hobby project with limited resources.
 
- 1. Download (and extract) Cppcheck source code.
- 2. Run:
-    ```
-    cd cppcheck/
-    virtualenv .env
-    .env/bin/pip install -r tools/donate-cpu-requirements.txt
-    .env/bin/python tools/donate-cpu.py
-    ```
+To monitor for stability, performance and regression we are analyzing the available source code of Debian packages with our latest development code
+and compare the results with the latest stable release.
 
-The script will analyse debian source code and upload the results to a cppcheck server. We need these results both to improve Cppcheck and to detect regressions.
+You can help with this by donating CPU time (1 core or as many as you like) and bandwidth.
 
-You can stop the script whenever you like with Ctrl C.
+It is as simple as running a shell script.
+
+You need a Linux (or WSL) shell (MacOS should work as well) with Python 3.x, Git and some common (development) tools (the script will tell you which) installed.
+
+NOTE: This will constantly and fully utilize the specified amount of cores while the script is running. It will also constantly fetch the sources which will be analyzed from a remote server (some of the packages are hundreds of megabytes in size).<br>
+If your resources and bandwidth are limited and/or would inflict certain costs on you (because of high energy prices or since you have to pay based on your data usage) you should probably not contribute. 
+
+```shell
+echo 'Thank you for providing resources and supporting the Cppcheck project!'
+
+# clone the 'main' branch of the official repository if it doesn't exist yet
+if [ ! -d cppcheck ]; then
+  git clone -b main https://github.com/danmar/cppcheck.git
+fi
+
+cd cppcheck
+
+while :
+do
+  # create a virtual environment to install the required Python dependencies (or update it)
+  python -m venv --upgrade --upgrade-deps .env
+  
+  # install or upgrade the Python dependencies
+  .env/bin/pip install -r tools/donate-cpu-requirements.txt --upgrade
+
+  # update to the latest source to make sure we run the latest client
+  git pull
+  
+  # run the client for a limited amount of packages (adjust -j to the amount of cores to use)
+  .env/bin/python tools/donate-cpu.py --max-packages=1000 -j1
+  
```

Comment:
possibly we should check exit codes here?

I'm not sure, but `donate-cpu.py` have some handling  of unstable network.
So, If it although exit with error, we need to handle this and maybe stop.

To be honest, I am a little afraid not only of the endless loop here, but especially auto updated script.

Reply:
> possibly we should check exit codes here?

Good point. Will take a look.

> I'm not sure, but `donate-cpu.py` have some handling of unstable network. 

It does. Depending on the operation it might also exit if it was not able to perform it.

> So, If it although exit with error, we need to handle this and maybe stop.

I do not think we need to handle this. If the network is not working the script will be restarted but won't do anything.

> To be honest, I am a little afraid not only of the endless loop here, but especially auto updated script.

If it doesn't encounter any issues it will run infinitely. And if it does it just won't do much and still run infinitely.

The auto-update is actually important. The client *always* needs to run the latest version otherwise they might produce wrong or missing data which caused a range of problems so far. Outdated clients also make it harder to remove deprecated functionality - see #5907.

Reply:
> it just won't do much
Why? It allow to execute arbitrary code for users, who has write access to this repo.
I understand that this is (endless loop) useful for running in isolated environment and check sometimes.
But I, as ordinary user, will prefer to know what I execute.

So, good that this part not in file and anyone can choose how to run it.

> I do not think we need to handle this. If the network is not working the script will be restarted but won't do anything.
I meant all other errors. which we most likely can't fix by just restarting script. May be using counter of sequenced errors will a good compromise.